### DISCUSSION TITLE
Couldn't install Gempy

### DISCUSSION THREAD

**Discussion by vmurugavel** [gempy-project/gempy#769] [Q&A]
I am having Python 3.11.2 in a windows 10 OS. I tried to install Gempy using pip. I got the following error at the end of the installation.

ERROR: Could not build wheels for pandas, which is required to install pyproject.toml-based projects

I then tried to install pandas older version (1.3.5) and ended up with an error.  Any advise on how to install Gempy as I am very curios about exploring it.

Thank you.

**Comment by AlexanderJuestel** [comment]
Hello,

What was the error you encountered with Pandas 1.3.5?

Cheers
Alex

**Reply by vmurugavel** [reply]
set build\lib.win-amd64-cpython-311\pandas/_version.py to '1.3.5'
      running build_ext
      building 'pandas._libs.algos' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

**Reply by vmurugavel** [reply]
I already have Microsoft Visual C++ 14.xxx. Not sure why pip installation doesn't recognise it.

**Reply by vmurugavel** [reply]
I have installed Microsoft Visual C++ and the above error is gone. 

When I import gempy in python, I got the following error.
>>> import gempy
No module named 'osgeo'
Not subsurface compatibility available
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python3\Lib\site-packages\gempy\__init__.py", line 25, in <module>
    from gempy.gempy_api import *
  File "C:\Python3\Lib\site-packages\gempy\gempy_api.py", line 17, in <module>
    from gempy.core.model import Project
  File "C:\Python3\Lib\site-packages\gempy\core\model.py", line 17, in <module>
    from gempy.core.interpolator import InterpolatorModel, InterpolatorGravity
  File "C:\Python3\Lib\site-packages\gempy\core\interpolator.py", line 1, in <module>
    from collections import Iterable
ImportError: cannot import name 'Iterable' from 'collections' (C:\Python3\Lib\collections\__init__.py)

----------------------------------------------------------------------------------------------

Then when I tried to install osgeo, I got the following error.

 Exception: In order to be able to run `from osgeo import gdal`,
      You were probably trying to install `gdal` by running `pip install osgeo`.
      Instead, you should either `pip install gdal` or replace `osgeo` with `gdal` in your requirements.

----------------------------------------------------------------------------------------------
when I tried to install pip install gdal, I got the following error.

error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.35.32215\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2

**Reply by Japhiolite** [reply]
There seems to be something messed up ... some digging suggested to force a re-install of `setuptools` in your python environment
`python -m pip install -U pip setuptools`

**Reply by vmurugavel** [reply]
I have re-installed setuptools as you have suggested. However I am getting the same error as I did before.

**Comment by 2651666L** [comment]
HI, there, I suggest you use anaconda (did you use it?). I have tried many times about the gempy installation, and i found that the better way for this is use anaconda to install it. You could set variety python versions in anaconda for gempy. In addition, running gempy may require a clean environment, and some unknown issues may cause problem in some processes. So, if possible, the best way I suggest is using anaconda in a virtual machine to install and use gempy, especially windows system.

**Comment by AlexanderJuestel** [comment]
@vmurugavel do you still have issues installing the latest version of GemPy using pip or conda?

**Reply by vmurugavel** [reply]
Thanks for getting back to me. I tried almost a year ago very hardly to make it work but I couldn't. I didn't try since then. Have you released a new version which works with latest python packages? If so I am been to try again.

**Reply by AlexanderJuestel** [reply]
Yes, we have released version 2.3 of GemPy which replaces theano with aesara and we have also fixed many dependency issues. So far, we only have minor complaints about installation issues. 

Have a try and let us know! 
Cheers Alex
